### PR TITLE
Use `dateparser` library for datetime parsing

### PR DESCRIPTION
**Pull request by Creepi0n**
Added hour, minute and second as optional parameters for --start and --end

### CODE REVIEW

File: command.py [hunk 1]
```diff
@@ -419,23 +419,27 @@ def human_timedelta(val):
...
     for char in val:
         if char == "/":
             parsing_i += 1
             continue
         if parsing_i > 2:
-            raise ParseError(f"Invalid date `{val}`. Expected format "
-                "`mm/dd/yyy`.")
+            if parsing_i > 5:
+                raise ParseError(f"Invalid date `{val}`. Expected format "
+                    "`mm/dd/yyyy` or `mm/dd/yyyy/hh/mm/ss`.")
```
Comment (Liam-DeVoe) [inline command.py]: `mm/dd/yyyy/hh/mm/ss` is definitely not a good format for this. Should be eg `7/12/2022 8:02:20 PM` as in the comment above. Ideally would also support 24 hour clocks when AM/PM isn't specified.

I'm guessing you had trouble getting `human_datetime` to be passed both arguments when they're separated with a space - you can achieve this by setting `convert_mode="together"` in the `Arg` definition of start and end. See --bounds for an example: https://github.com/tybug/snitchvisbot/blob/5566d5438daad13ed0e989c4c62aac204e76d38a/main.py#L679

File: command.py [hunk 2]
```diff
@@ -419,23 +419,27 @@ def human_timedelta(val):
...
         if char == " ":
             break
         parts[parsing_i] = parts[parsing_i] + char
 
     # make sure we've parsed at least one char for all of month/day/year
-    if parsing_i != 2:
-        raise ParseError(f"Invalid date `{val}`. Expected format `mm/dd/yyyy`.")
+    if parsing_i != 5:
+        if parsing_i != 2:
```
Comment (Liam-DeVoe) [inline command.py]: combine these two conditions

File: command.py [hunk 3]
```diff
@@ -458,7 +462,33 @@ def human_datetime(val):
                 "digits.")
     except ValueError:
         raise ParseError(f"Invalid year `{parts[2]}`.")
+    if len(parts[3]) == 0:
+        hour=0
```
Comment (Liam-DeVoe) [inline command.py]: spaces between equals please

`hour = 0`

**Review by Liam-DeVoe — CHANGES_REQUESTED** [review]

**Comment by Gjum** [comment]
Why write such a complex parser for a simple date? Use [datetime.strptime](https://docs.python.org/3/library/datetime.html#datetime.datetime.strptime) for a builtin parser (use try/except to allow multiple formats), or [pandas.to_datetime](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.to_datetime.html) if you want to allow many flexible formats.

**Comment by Liam-DeVoe** [comment]
Mostly for nice error messages. strptime errors can be pretty inscrutable if you don't know format specifiers (`%Y/%m/%d` might be understandable to most, but `%Y/%m/%d %H:%M:%S %p` significantly less so with the %p).

I could just return a generic error I guess, eg `Invalid value; must match eg 2022/07/03 06:03:20 PM, AM/PM is optional if using 24 hour clock`, but I dunno I think that's more confusing than pointing out the exact error location.

Pandas is an interesting suggestion, but I would rather write a parser like this than use such a heavy library just for datetime parsing.

**Comment by Liam-DeVoe** [comment]
I've decided that @Gjum was right and maintaining our own timestamp parser is (probably) more trouble than it's worth. It's nice having granular control over formats and error messages, but supporting all reasonable formats is a huge pain. 

I've offloaded this to the `dateparser` library for now, with the added bonus that things like `three weeks ago` are now valid inputs everywhere a datetime is accepted. The formats that this PR was initially opened to support are also accepted now as well.

**Pull request merged**